gateway/src/cluster/impl.rs

```diff
@@ -92,7 +92,7 @@ struct ClusterRef {
     config: ClusterConfig,
     shard_from: u64,
     shard_to: u64,
-    shards: Mutex<HashMap<u64, Shard>>,
```

Comment:
I am not too sure about this change, would it not likely be just as good to have a sync mutex here? Instead of using a DashMap.

Reply:
addressed